app/src/main/java/it/niedermann/owncloud/notes/edit/NotePreviewFragment.java

```diff
@@ -114,6 +114,7 @@ protected void onNoteLoaded(Note note) {
         super.onNoteLoaded(note);
         noteLoaded = true;
         registerInternalNoteLinkHandler();
+        binding.singleNoteContent.setMarkdownImageUrlPrefix(repo.getAccountById(note.getAccountId()).getUrl()+"/index.php/apps/notes/notes/"+note.getRemoteId()+"/attachment?path=");
```

Comment:
The internal API is used by the web app, only. It has no public contract, so it can be changed at any time without information to third-party developers. It is fitted to the demands of the web app and not to the demands of third-party apps like mobile apps (e.g. which have an offline mode).

In contrast to this, the public API has a [public documentation](https://github.com/nextcloud/notes/blob/master/docs/api/README.md) as well as [continuous integration tests](https://github.com/nextcloud/notes/tree/master/tests/api), it changes very seldom, changes must be backwards-compatible (create an incompatible new major API version should be used with caution) and it is fitted to the needs of mobile apps.